### ISSUE TITLE
Generated code can't pass staticcheck

### ISSUE THREAD

**Issue by sunyakun** [cloudwego/thriftgo#26]
#### Description

thriftgo generated code can't pass `staticcheck`, so we must write an extra config file `staticcheck.conf` to ignore some checks, but `staticcheck` will ignore all golang code files under the same directory tree which the config file placed

#### Reproduction

assume we have a thrift file named 'foo.thrift' and below is the content

```thrift
namespace golang foo

struct Foo {
    1: required string name,
    2: required i64 age,
}
```

execute shell commands below

```shell
go mod init code.byted.org/demo
thriftgo -g go -o ./ foo.thrift
go mod tidy
go mod edit -replace  github.com/apache/thrift=github.com/apache/thrift@v0.13.0
go mod tidy

staticcheck ./...
```

and will get the output

```shell
foo/foo.go:113:67: error strings should not be capitalized (ST1005)
```

#### Recommandation

1. add `//lint:ignore` directive to generator/golang/templates/struct.go::StructLikeRead

``` golang
RequiredFieldNotSetError:
	//lint:ignore ST1005 {some reasons}
	return thrift.NewTProtocolExceptionWithType(thrift.INVALID_DATA, fmt.Errorf("Required field %s is not set", fieldIDToName_ModelSync[fieldId]))
```

2. use lowercase word instead

```golang
return thrift.NewTProtocolExceptionWithType(thrift.INVALID_DATA, fmt.Errorf("required field %s is not set", fieldIDToName_ModelSync[fieldId]))
```

#### Reference

- https://staticcheck.io/docs/configuration/#line-based-linter-directives

**Comment by lsjbd** [comment]
That's awkward. Files generated by thriftgo all contains a header comment `// Code generated by thriftgo (x.y.z). DO NOT EDIT.` and staticcheck should ignore them as most linters do. Can you check if this is an issue of staticcheck ?

**Comment by sunyakun** [comment]
This is a known [issue](https://github.com/dominikh/go-tools/issues/429#issuecomment-477069038) of staticcheck, but the author recommends 'generator' to fix the problems, and I also find a code review comment about error string: https://github.com/golang/go/wiki/CodeReviewComments#error-strings, so does this is a bad smell of thriftgo or there have some reasons to use capitalized in error string?

**Comment by lsjbd** [comment]
`thrift.INVALID_DATA` is apparently a contant defined in the apache thrift go library. It is **not** a generated symbol. So there are two solutions available: PR for apache thrift to change all their contant definitions to meet golang conventions then we update thriftgo to follow up; or simply push the author of staticcheck to fix its issue.

After all, staticcheck does not follow the common way for ignoring generated files but asks all potential generators to meet its requirement. I think this is irrational.

And, `DO NOT EDIT` is recommended by the go officials. See https://github.com/golang/go/issues/41196.

**Comment by lsjbd** [comment]
OK, we will fix the error string issue in the next version.

But after all, staticcheck does not follow the common way for ignoring generated files but asks all potential generators to meet its requirement. I think this is irrational.

And, `DO NOT EDIT` is recommended by the go officials. See https://github.com/golang/go/issues/41196.

**Comment by lsjbd** [comment]
This issue has been solved in v0.1.4.
